js/ui/extension.js

```diff
@@ -181,7 +181,7 @@ Extension.prototype = {
 
         const finishLoad = () => {
             // Many xlets still use appletMeta/deskletMeta to get the path
-            type.legacyMeta[uuid] = {path: this.meta.path};
+            type.legacyMeta[uuid] = Object.assign({}, this.meta);
```

Comment:
Can you please point to a piece of code that warrants this use case? We are storing the full metadata in every extension and xlet instance if we do this. Its inefficient. There is no discernible difference between the time _init is called and whatever you want to do outside the xlet function constructor. After our conversation in #6878 I'm still convinced this is more about principal, not what's best for Cinnamon technically.